Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/InputActionsEditorWindowUtils.cs

```diff
@@ -13,18 +15,35 @@ internal class InputActionsEditorWindowUtils
         ? AssetDatabase.LoadAssetAtPath<StyleSheet>(InputActionsEditorConstants.PackagePath + InputActionsEditorConstants.ResourcesPath + "/InputAssetEditorDark.uss")
         : AssetDatabase.LoadAssetAtPath<StyleSheet>(InputActionsEditorConstants.PackagePath + InputActionsEditorConstants.ResourcesPath + "/InputAssetEditorLight.uss");
 
-        public enum ConfirmSaveChangesDialogResult
+        // Similar to InputActionAsset.WriteFileJson but excludes the name
+        [Serializable]
+        private struct WriteFileJsonNoName
         {
-            Save = 0,
-            Cancel = 1,
-            DontSave = 2
+            public InputActionMap.WriteMapJson[] maps;
+            public InputControlScheme.SchemeJson[] controlSchemes;
         }
 
-        public static ConfirmSaveChangesDialogResult ConfirmSaveChanges(string path)
+        // Similar to InputActionAsset.ToJson() but converts to JSON excluding the name property and any additional JSON
+        // content that may be part of the file not recognized as required data.
+        public static string ToJsonWithoutName(InputActionAsset asset)
```

Comment:
Are these actually used and do we really need them ?

Reply:
Yes they are used to convert an asset to JSON but exclude the name field. It is used for comparing two assets content to update dirty flags. @timkeo made a change previously which aligns JSON and SerializedObject names when desynchronised, but there is no need for editor to go dirty if that is the only change, similar to us discarding other "augmented" fields in json that users might enter. This also allowed removing a previous workaround for comparing whether an asset was empty or not that was a bit clunky.

Reply:
I noticed now however that the Read version is not used anymore (refactored that one away) so I removed it in a coming commit.